android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -2163,10 +2148,14 @@ public void setDrmLicenseHeader(String[] header){
     public void setAnalyticsMeta(ReadableMap analyticsData) {
         this.analyticsMeta = analyticsData;
 
-        if (analyticsData != null && analyticsData.getBoolean("contentIsLive") && srcChanged) {
-            if (player != null && analyticsData != null && youboraPlugin == null && contentId != analyticsData.getString("contentId")) {
-                initialiseYoubora();
-            }
+        if (player != null && youboraPlugin == null && (analyticsMeta != null && analyticsMeta.getBoolean("contentIsLive") && contentId != analyticsMeta.getString("contentId"))) {
```

Comment:
it seems like it would be an error if any of these are null when setAnalyticsMeta is called, whereas this will just silently do nothing.

Reply:
`contentId != analyticsMeta.getString("contentId")` you need to use .equals to compare strings in Java. Suspect you might not have tested this clause?

Reply:
If any of the condition is null, it should do nothing, like player, if player is not init youbora can't be loaded,and this is only for live, so this will check contentIsLive. I will update use .equals

Reply:
why would you call setAnalyticsMeta if `player` or  `analyticsData` is null?

Reply:
> and this is only for live

Why don't we have youbora for VOD?

Reply:
we have youbora for vod, it's in the init player logic, we init player then init youbora, for vod we don't change analyticMeta during play. For live we change channel or play to the next slot so this function need to be called

Reply:
I'll be honest - it's pretty much impossible to follow this code, which I think leads the continual number of regressions.

Reply:
if player is null this simply do nothing, it won't throw exception. If analyticMeta is null, it will stop youbora, that is how we control youbora. For youbora a full life cycle is init -> monitor -> stop, we must firestop to make it record the stop time. If we continue to play in a live tv after slot finish we need to stop current youbora and restart a new one, before the stop function is tied to the source change, it works fine for vod or change channel, but slot change don't change source thus it's broken.

Reply:
This code don't work alone it will work with front end logic, the code itself is to set live tv youbora depends on analyticMeta change instead of other changes

Reply:
I think you're missing my point, you can't tell any of the above from reading the code. The overall implementation is simply spaghetti (which I realise predates this change, but this change just doubles down on): 
- youbora is be initialised from totally different places depending on whether its live or vod. We're using (abusing) same code in entirely different ways rather than properly factoring out VOD vs live playback (which unfortunately have ended up wildy different).
- we use magic strings inside the analytics metadata to decide if its live or not live
- we rely on a "setAnalyticsMeta" call to trigger init/stop calls, which is counter-intuitive (obviously this method does a lot more than just populate analytics metadata) and relies on this method being called at just the right time. 

There are so many ways this can go wrong. There are no tests. There are million different broken/odd/buggy states possible. There are no comments explaining the lifecycle of the player, which is particularly bad because the code itself is not self-describing ("call setAnalyticsMetadata with null to trigger stop"). This code makes no sense to another developer new to the area. Making what seems like a harmless change can lots of surprising side effects.

Stepping back, it sounds like we're abandoning the idea of the single video player, since it can't be made to work in skygo, so perhaps this will remove some of the hacks we've put in here.

Reply:
Youbora live and vod initialized from different place is because for vod we already have all the info before we start to watch, we can prepare the analyticMeta and source together and init them together. Also since vod don't change content during play youbora will stop when video stops. But for live we fetch slot detail after we get into the channel, then prepare the analyticMeta which sometime slower than the source, so during the init process youbora could be not inited with player also another thing is during live play we could change channel we could change slot that will change the analyticMeta but player still playing that's why I use setAnalyticMeta fucntion to control youbora.
So for vod, youbora init with player, stop with player, but for live I decouple the youbora with player, make the youbora life cycle depend on analyticMeta change , because a slot change is not source change it not a 'change' to player, only this analyticMeta prop will change.

Reply:
Just to close off this thread - we should not have completely different APIs or interaction models for the player for vod or live. If live has a more complex needs (initialising youbora after source load, re-initialising youbora during playback as slot changes), then we create a clear API that properly supports that, which enforces if/when it can be called during the lifecycle (not done currently), and use the same one for VOD.

This would get rid of these hacky is/is-not `getBoolean("contentIsLive")` calls (see https://en.wikipedia.org/wiki/Magic_string), because the player wouldn't need to know about this distinction.